src/argilla/client/feedback/dataset/remote/base.py

```diff
@@ -155,12 +160,13 @@ def __getitem__(self, key: Union[slice, int]) -> Union[RemoteFeedbackRecord, Lis
                 id=self._dataset_id,
                 offset=offset,
                 limit=limit,
+                **filters,
             ).parsed
             records.extend([self.__parse_record(record) for record in fetched_records.items])
         return records[0] if isinstance(key, int) else records
 
     @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
-    def __iter__(self) -> Iterator[RemoteFeedbackRecord]:
+    def _iter_records(self, filters: Dict[str, Any] = {}) -> Iterator[RemoteFeedbackRecord]:
```

Comment:
```suggestion
    def _iter_records(self, filters: Dict[str, Any] = {}) -> Iterator[RemoteFeedbackRecordBase]:
```

Reply:
Same here, there's no `RemoteFeedbackRecordBase` class, and all these classes return `RemoteFeedbackRecord` as that's the same on both `RemoteFeedbackDataset` and `FilteredRemoteFeedbackDataset`